crates/compiler/builtins/roc/Hash.roc

```diff
@@ -73,12 +75,38 @@ Hasher has
     ## accumulated hash state.
     complete : a -> U64 | a has Hasher
 
+    ## Resets the internal state of a hasher
+    ## The hasher should still have all the same parameters and seeds.
```

Comment:
Separately, I'd suggest we explain in the doc comment here that this is useful for unordered hashing.

Reply:
function removed.